### PR TITLE
SDK Cocoa LumberJack File Logger code

### PR DESCRIPTION
**Pull request by trooper2013**
This PR moves all the SFSDKLogger code into this repo.

-  Relocation of previous code. 
- Comply with SFLogging protocol.
- Mapped back and forth the levels between SFLogLevel and DDLogLevel
- Left some of the class method api(s) on the SFSDKLogger the same (using DDLogLevel).  We could dod the same for the instance methods too. (needs your input)
- Cocoa Lumberjack is added as a git submodule.
- SalesforceSDKCommon is pulled from cocoapods. 
 
TBD - (Dependency)
* Clean up pod file.
* Add a podspec for this module.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[submodule "external/CocoaLumberjack"]
+	path = external/CocoaLumberjack
+	url = https://github.com/forcedotcom/CocoaLumberjack
+[submodule "external/SalesforceMobileSDK-iOS"]
```
Comment (wmathurin) [inline .gitmodules]: That's going to be interesting with SalesforceMobileSDK-iOS-Hybrid
It also has a submodule pointing to SalesforceMobileSDK-iOS.
All you need from SalesforceMobileSDK-iOS is the common lib right?

File: SalesforceFileLoggerTestApp/main.m [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//
```
Comment (wmathurin) [inline SalesforceFileLoggerTestApp/main.m]: Use standard copyright

File: .gitmodules (hunk 1 above)
Comment (wmathurin) [reply .gitmodules]: Should we move that code to SalesforceMobileSDK-iOS-Hybrid? Or will it be needed elsewhere?

Comment (wmathurin) [reply .gitmodules]: Could we have a build script to create the .a for cordova?

**Comment by bhariharan** [comment]
@wmathurin Are we going to move this to the Hybrid repo? Can this be closed?

**Comment by wmathurin** [comment]
@bhariharan it's moving to the SalesforceMobileSDK-iOS-Hybrid repo.
Here is the PR: https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid/pull/4

**Comment by wmathurin** [comment]
Closing this PR - we can probably get rid of the repo as well.

**Comment by bhariharan** [comment]
Cool - I'll delete the repo.

**Pull request closed without merging**